app/controllers/admin/deals_controller.rb

```diff
@@ -0,0 +1,51 @@
+class Admin::DealsController < Admin::BaseController
+  def index
+    @deals = Deal.all
+  end
+
+  def new
+    @deal  = Deal.new
+  end
+
+  def edit
+    @deal = Deal.find(params[:id])
```

Comment:
@jynti Please move this to before action and make sure you do not repeat this mistake.